### PR TITLE
[7/8] ENH: upgrade the MERFISH notebook to use IntensityTable and Codebook …

### PR DESCRIPTION
**Pull request by ambrosejcarr**
- Upgrade the MERFISH notebook to use IntensityTable and Codebook

Note: The correlation between the new methods and the expected results is lower (0.9) than it was previously. This is likely due to selection of filtering parameters and still needs debugging. However, this is significantly "more working" than the existing notebook, so I think is worth PRing and getting merged so that users are less confused about broken notebooks. 

Also worth noting that at this point, the text in _most_ notebooks is probably significantly out of date and needs an upgrade. :-)

### CODE REVIEW

**Comment by ambrosejcarr** [comment]
Closes #308

File: notebooks/MERFISH_Pipeline_-_U2O2_Cell_Culture_-_1_FOV.py [hunk 1]
```diff
@@ -184,41 +184,15 @@
...
-[f for f in dir(decoded) if not f.startswith('_')]
-# EPY: END code
-
-# EPY: START markdown
-# Spot attributes are stored as skimage RegionProperties attributes
-# EPY: END markdown
-
-# EPY: START code
-decoded.spot_props[:3]
+spot_intensities, region_properties, label_image, intensity_image, decoded_image = psd.find(s.image)
```
Comment (dganguli) [inline notebooks/MERFISH_Pipeline_-_U2O2_Cell_Culture_-_1_FOV.py]: this is a pretty different API than when you use the GaussianSpotDetector in the ISS pipeline: 

```
p = SpotFinder.GaussianSpotDetector(
    min_sigma=min_sigma,
    max_sigma=max_sigma,
    num_sigma=num_sigma,
    threshold=threshold,
    blobs_stack=blobs_stack,
    measurement_type='mean',
)

# blobs = dots; define the spots in the dots image, but then find them again in the stack.
intensities = p.find(filtered_stack)
```


Perhaps a simpler API would be that all find methods return a 'spot_intensities' table AND an optional named dictionary of key value pairs.

File: notebooks/MERFISH_Pipeline_-_U2O2_Cell_Culture_-_1_FOV.py [hunk 2]
```diff
@@ -45,7 +44,7 @@
 
 # EPY: START code
 # show all hybridization rounds of channel 0
-s.image.show_stack({Indices.CH: 0})
+s.image.show_stack({Indices.CH.value: 0}, rescale=False)
```
Comment (ttung) [inline notebooks/MERFISH_Pipeline_-_U2O2_Cell_Culture_-_1_FOV.py]: do you need the `.value` part?  i don't think you should need that...

File: notebooks/MERFISH_Pipeline_-_U2O2_Cell_Culture_-_1_FOV.py [hunk 3]
```diff
@@ -230,27 +204,26 @@
...
+result_counts = pd.Series(counts, index=genes)
+
+tmp = pd.concat([result_counts, benchmark_counts], join='inner', axis=1).values
+
+r = np.corrcoef(tmp[:,1], tmp[:,0])[0,1]
 x = np.linspace(50, 2000)
-plt.scatter(tmp[:,1],tmp[:,0], 50,zorder=2)
-plt.plot(x,x,'-k',zorder=1)
+f, ax = plt.subplots(figsize=(6, 6))
+ax.scatter(tmp[:,1],tmp[:,0], 50,zorder=2)
```
Comment (ttung) [inline notebooks/MERFISH_Pipeline_-_U2O2_Cell_Culture_-_1_FOV.py]: style nits: should have space after comma.

File: notebooks/MERFISH_Pipeline_-_U2O2_Cell_Culture_-_1_FOV.py [hunk 4]
```diff
@@ -230,27 +204,26 @@
...
+ax.plot(x,x,'-k',zorder=1)
 
 plt.xlabel('Gene copy number Benchmark')
 plt.ylabel('Gene copy number Starfish')
 plt.xscale('log')
 plt.yscale('log')
-plt.title('r = {}'.format(r))
-
-sns.despine(offset=2)
+plt.title('r = {}'.format(r));
```
Comment (ttung) [inline notebooks/MERFISH_Pipeline_-_U2O2_Cell_Culture_-_1_FOV.py]: plt.title(f"r = {r}")

File: notebooks/MERFISH_Pipeline_-_U2O2_Cell_Culture_-_1_FOV.py [hunk 5]
```diff
@@ -133,7 +133,7 @@
 
 for indices in s.image._iter_indices():
     data = s.image.get_slice(indices)[0]
-    scaled = data / scale_factors[indices[Indices.HYB], indices[Indices.CH]]
+    scaled = data / scale_factors[indices[Indices.HYB.value], indices[Indices.CH.value]]
```
[the remark this answers is not in the document]
Comment (ambrosejcarr) [reply notebooks/MERFISH_Pipeline_-_U2O2_Cell_Culture_-_1_FOV.py]: Not anymore. I'll get them in #340, though.

**Comment by ttung** [comment]
Can you refresh the PR runs?

**Pull request merged**